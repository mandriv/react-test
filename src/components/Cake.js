import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Item, Dropdown, Responsive } from 'semantic-ui-react';
import EditCakeModal from '../components/EditCakeModal';

/*
  CakeSegment description
*/

export default class Cake extends Component {

  state = { editMode: false }

  render() {
    const { editMode } = this.state;
    const { image, title, desc } = this.props.item;
    return (
      <Item>
        <Item.Image size="small" src={image} />
        <Item.Content>
          <Item.Header className="capitalize">
            {title}
          </Item.Header>
          <Item.Meta className="capitalize">
            {desc}
          </Item.Meta>
          {/* Mobile only */}
          <Responsive
            {...Responsive.onlyMobile}
            as={Item.Extra}
          >
            <Dropdown
              icon="settings"
              floated="right"
              className="top left pointing dropdown"
            >
              <Dropdown.Menu>
                <Dropdown.Item
                  onClick={() => this.setState({ editMode: true })}
                >
                  Edit
                </Dropdown.Item>
                <Dropdown.Item
                  onClick={() => this.props.onRemove(this.props.item)}
                >
                  Delete
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </Responsive>
        </Item.Content>
        {/* Non mobile */}
        <Responsive
          minWidth={Responsive.onlyTablet.minWidth}
        >
          <Dropdown
            icon="settings"
            floated="right"
            className="top right pointing dropdown"
          >
            <Dropdown.Menu>
              <Dropdown.Item
                onClick={() => this.setState({ editMode: true })}
              >
                Edit
              </Dropdown.Item>
              <Dropdown.Item
                onClick={() => this.props.onRemove(this.props.item)}
              >
                Delete
              </Dropdown.Item>
            </Dropdown.Menu>
          </Dropdown>
        </Responsive>
        <EditCakeModal
          cake={this.props.item}
          open={editMode}
          closeIcon
          closeOnDocumentClick
          onClose={() => this.setState({ editMode: false })}
          onSubmit={newCake => this.props.onEdit(newCake)}
        />
      </Item>
    );
  }

}

Cake.propTypes = {
  item: PropTypes.object.isRequired,
  onRemove: PropTypes.func,
  onEdit: PropTypes.func,
};

Cake.defaultProps = {
  onRemove: () => null,
  onEdit: () => null,
};
