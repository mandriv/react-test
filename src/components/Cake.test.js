import React from 'react';
import { shallow } from 'enzyme';
import Cake from './Cake';

describe('Component: <Cake>', () => {
  it('Renders <Cake>', () => {
    const cake = {
      title: 'Testy Cake',
      desc: 'description of test cake',
      image: 'https://picsum.photos/400',
    };
    const cakeComponent = shallow(<Cake item={cake} />);
    expect(cakeComponent).toMatchSnapshot();
  });
});
