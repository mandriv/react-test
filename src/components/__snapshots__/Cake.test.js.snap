// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Cake> Renders <Cake> 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Cake
    item={
        Object {
            "desc": "description of test cake",
            "image": "https://picsum.photos/400",
            "title": "Testy Cake",
          }
    }
    onEdit={[Function]}
    onRemove={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ItemImage
          size="small"
          src="https://picsum.photos/400"
/>,
        <ItemContent>
          <ItemHeader
                    className="capitalize"
          >
                    Testy Cake
          </ItemHeader>
          <ItemMeta
                    className="capitalize"
          >
                    description of test cake
          </ItemMeta>
          <Responsive
                    as={[Function]}
                    maxWidth={767}
                    minWidth={320}
          >
                    <Dropdown
                              additionLabel="Add "
                              additionPosition="top"
                              className="top left pointing dropdown"
                              closeOnBlur={true}
                              deburr={false}
                              floated="right"
                              icon="settings"
                              minCharacters={1}
                              noResultsMessage="No results found."
                              openOnFocus={true}
                              renderLabel={[Function]}
                              searchInput="text"
                              selectOnBlur={true}
                              selectOnNavigation={true}
                    >
                              <DropdownMenu>
                                        <DropdownItem
                                                  onClick={[Function]}
                                        >
                                                  Edit
                                        </DropdownItem>
                                        <DropdownItem
                                                  onClick={[Function]}
                                        >
                                                  Delete
                                        </DropdownItem>
                              </DropdownMenu>
                    </Dropdown>
          </Responsive>
</ItemContent>,
        <Responsive
          minWidth={768}
>
          <Dropdown
                    additionLabel="Add "
                    additionPosition="top"
                    className="top right pointing dropdown"
                    closeOnBlur={true}
                    deburr={false}
                    floated="right"
                    icon="settings"
                    minCharacters={1}
                    noResultsMessage="No results found."
                    openOnFocus={true}
                    renderLabel={[Function]}
                    searchInput="text"
                    selectOnBlur={true}
                    selectOnNavigation={true}
          >
                    <DropdownMenu>
                              <DropdownItem
                                        onClick={[Function]}
                              >
                                        Edit
                              </DropdownItem>
                              <DropdownItem
                                        onClick={[Function]}
                              >
                                        Delete
                              </DropdownItem>
                    </DropdownMenu>
          </Dropdown>
</Responsive>,
        <EditCakeModal
          cake={
                    Object {
                              "desc": "description of test cake",
                              "image": "https://picsum.photos/400",
                              "title": "Testy Cake",
                            }
          }
          closeIcon={true}
          closeOnDocumentClick={true}
          onClose={[Function]}
          onSubmit={[Function]}
          open={false}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "size": "small",
          "src": "https://picsum.photos/400",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ItemHeader
              className="capitalize"
>
              Testy Cake
</ItemHeader>,
            <ItemMeta
              className="capitalize"
>
              description of test cake
</ItemMeta>,
            <Responsive
              as={[Function]}
              maxWidth={767}
              minWidth={320}
>
              <Dropdown
                            additionLabel="Add "
                            additionPosition="top"
                            className="top left pointing dropdown"
                            closeOnBlur={true}
                            deburr={false}
                            floated="right"
                            icon="settings"
                            minCharacters={1}
                            noResultsMessage="No results found."
                            openOnFocus={true}
                            renderLabel={[Function]}
                            searchInput="text"
                            selectOnBlur={true}
                            selectOnNavigation={true}
              >
                            <DropdownMenu>
                                          <DropdownItem
                                                        onClick={[Function]}
                                          >
                                                        Edit
                                          </DropdownItem>
                                          <DropdownItem
                                                        onClick={[Function]}
                                          >
                                                        Delete
                                          </DropdownItem>
                            </DropdownMenu>
              </Dropdown>
</Responsive>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Testy Cake",
              "className": "capitalize",
            },
            "ref": null,
            "rendered": "Testy Cake",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "description of test cake",
              "className": "capitalize",
            },
            "ref": null,
            "rendered": "description of test cake",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": [Function],
              "children": <Dropdown
                additionLabel="Add "
                additionPosition="top"
                className="top left pointing dropdown"
                closeOnBlur={true}
                deburr={false}
                floated="right"
                icon="settings"
                minCharacters={1}
                noResultsMessage="No results found."
                openOnFocus={true}
                renderLabel={[Function]}
                searchInput="text"
                selectOnBlur={true}
                selectOnNavigation={true}
>
                <DropdownMenu>
                                <DropdownItem
                                                onClick={[Function]}
                                >
                                                Edit
                                </DropdownItem>
                                <DropdownItem
                                                onClick={[Function]}
                                >
                                                Delete
                                </DropdownItem>
                </DropdownMenu>
</Dropdown>,
              "maxWidth": 767,
              "minWidth": 320,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "additionLabel": "Add ",
                "additionPosition": "top",
                "children": <DropdownMenu>
                  <DropdownItem
                                    onClick={[Function]}
                  >
                                    Edit
                  </DropdownItem>
                  <DropdownItem
                                    onClick={[Function]}
                  >
                                    Delete
                  </DropdownItem>
</DropdownMenu>,
                "className": "top left pointing dropdown",
                "closeOnBlur": true,
                "deburr": false,
                "floated": "right",
                "icon": "settings",
                "minCharacters": 1,
                "noResultsMessage": "No results found.",
                "openOnFocus": true,
                "renderLabel": [Function],
                "searchInput": "text",
                "selectOnBlur": true,
                "selectOnNavigation": true,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <DropdownItem
                      onClick={[Function]}
>
                      Edit
</DropdownItem>,
                    <DropdownItem
                      onClick={[Function]}
>
                      Delete
</DropdownItem>,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Edit",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Edit",
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "children": "Delete",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Delete",
                    "type": [Function],
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Dropdown
            additionLabel="Add "
            additionPosition="top"
            className="top right pointing dropdown"
            closeOnBlur={true}
            deburr={false}
            floated="right"
            icon="settings"
            minCharacters={1}
            noResultsMessage="No results found."
            openOnFocus={true}
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
>
            <DropdownMenu>
                        <DropdownItem
                                    onClick={[Function]}
                        >
                                    Edit
                        </DropdownItem>
                        <DropdownItem
                                    onClick={[Function]}
                        >
                                    Delete
                        </DropdownItem>
            </DropdownMenu>
</Dropdown>,
          "minWidth": 768,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "additionLabel": "Add ",
            "additionPosition": "top",
            "children": <DropdownMenu>
              <DropdownItem
                            onClick={[Function]}
              >
                            Edit
              </DropdownItem>
              <DropdownItem
                            onClick={[Function]}
              >
                            Delete
              </DropdownItem>
</DropdownMenu>,
            "className": "top right pointing dropdown",
            "closeOnBlur": true,
            "deburr": false,
            "floated": "right",
            "icon": "settings",
            "minCharacters": 1,
            "noResultsMessage": "No results found.",
            "openOnFocus": true,
            "renderLabel": [Function],
            "searchInput": "text",
            "selectOnBlur": true,
            "selectOnNavigation": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <DropdownItem
                  onClick={[Function]}
>
                  Edit
</DropdownItem>,
                <DropdownItem
                  onClick={[Function]}
>
                  Delete
</DropdownItem>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Edit",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Edit",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "Delete",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Delete",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "cake": Object {
            "desc": "description of test cake",
            "image": "https://picsum.photos/400",
            "title": "Testy Cake",
          },
          "closeIcon": true,
          "closeOnDocumentClick": true,
          "onClose": [Function],
          "onSubmit": [Function],
          "open": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ItemImage
            size="small"
            src="https://picsum.photos/400"
/>,
          <ItemContent>
            <ItemHeader
                        className="capitalize"
            >
                        Testy Cake
            </ItemHeader>
            <ItemMeta
                        className="capitalize"
            >
                        description of test cake
            </ItemMeta>
            <Responsive
                        as={[Function]}
                        maxWidth={767}
                        minWidth={320}
            >
                        <Dropdown
                                    additionLabel="Add "
                                    additionPosition="top"
                                    className="top left pointing dropdown"
                                    closeOnBlur={true}
                                    deburr={false}
                                    floated="right"
                                    icon="settings"
                                    minCharacters={1}
                                    noResultsMessage="No results found."
                                    openOnFocus={true}
                                    renderLabel={[Function]}
                                    searchInput="text"
                                    selectOnBlur={true}
                                    selectOnNavigation={true}
                        >
                                    <DropdownMenu>
                                                <DropdownItem
                                                            onClick={[Function]}
                                                >
                                                            Edit
                                                </DropdownItem>
                                                <DropdownItem
                                                            onClick={[Function]}
                                                >
                                                            Delete
                                                </DropdownItem>
                                    </DropdownMenu>
                        </Dropdown>
            </Responsive>
</ItemContent>,
          <Responsive
            minWidth={768}
>
            <Dropdown
                        additionLabel="Add "
                        additionPosition="top"
                        className="top right pointing dropdown"
                        closeOnBlur={true}
                        deburr={false}
                        floated="right"
                        icon="settings"
                        minCharacters={1}
                        noResultsMessage="No results found."
                        openOnFocus={true}
                        renderLabel={[Function]}
                        searchInput="text"
                        selectOnBlur={true}
                        selectOnNavigation={true}
            >
                        <DropdownMenu>
                                    <DropdownItem
                                                onClick={[Function]}
                                    >
                                                Edit
                                    </DropdownItem>
                                    <DropdownItem
                                                onClick={[Function]}
                                    >
                                                Delete
                                    </DropdownItem>
                        </DropdownMenu>
            </Dropdown>
</Responsive>,
          <EditCakeModal
            cake={
                        Object {
                                    "desc": "description of test cake",
                                    "image": "https://picsum.photos/400",
                                    "title": "Testy Cake",
                                  }
            }
            closeIcon={true}
            closeOnDocumentClick={true}
            onClose={[Function]}
            onSubmit={[Function]}
            open={false}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": "small",
            "src": "https://picsum.photos/400",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ItemHeader
                className="capitalize"
>
                Testy Cake
</ItemHeader>,
              <ItemMeta
                className="capitalize"
>
                description of test cake
</ItemMeta>,
              <Responsive
                as={[Function]}
                maxWidth={767}
                minWidth={320}
>
                <Dropdown
                                additionLabel="Add "
                                additionPosition="top"
                                className="top left pointing dropdown"
                                closeOnBlur={true}
                                deburr={false}
                                floated="right"
                                icon="settings"
                                minCharacters={1}
                                noResultsMessage="No results found."
                                openOnFocus={true}
                                renderLabel={[Function]}
                                searchInput="text"
                                selectOnBlur={true}
                                selectOnNavigation={true}
                >
                                <DropdownMenu>
                                                <DropdownItem
                                                                onClick={[Function]}
                                                >
                                                                Edit
                                                </DropdownItem>
                                                <DropdownItem
                                                                onClick={[Function]}
                                                >
                                                                Delete
                                                </DropdownItem>
                                </DropdownMenu>
                </Dropdown>
</Responsive>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Testy Cake",
                "className": "capitalize",
              },
              "ref": null,
              "rendered": "Testy Cake",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "description of test cake",
                "className": "capitalize",
              },
              "ref": null,
              "rendered": "description of test cake",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": [Function],
                "children": <Dropdown
                  additionLabel="Add "
                  additionPosition="top"
                  className="top left pointing dropdown"
                  closeOnBlur={true}
                  deburr={false}
                  floated="right"
                  icon="settings"
                  minCharacters={1}
                  noResultsMessage="No results found."
                  openOnFocus={true}
                  renderLabel={[Function]}
                  searchInput="text"
                  selectOnBlur={true}
                  selectOnNavigation={true}
>
                  <DropdownMenu>
                                    <DropdownItem
                                                      onClick={[Function]}
                                    >
                                                      Edit
                                    </DropdownItem>
                                    <DropdownItem
                                                      onClick={[Function]}
                                    >
                                                      Delete
                                    </DropdownItem>
                  </DropdownMenu>
</Dropdown>,
                "maxWidth": 767,
                "minWidth": 320,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "additionLabel": "Add ",
                  "additionPosition": "top",
                  "children": <DropdownMenu>
                    <DropdownItem
                                        onClick={[Function]}
                    >
                                        Edit
                    </DropdownItem>
                    <DropdownItem
                                        onClick={[Function]}
                    >
                                        Delete
                    </DropdownItem>
</DropdownMenu>,
                  "className": "top left pointing dropdown",
                  "closeOnBlur": true,
                  "deburr": false,
                  "floated": "right",
                  "icon": "settings",
                  "minCharacters": 1,
                  "noResultsMessage": "No results found.",
                  "openOnFocus": true,
                  "renderLabel": [Function],
                  "searchInput": "text",
                  "selectOnBlur": true,
                  "selectOnNavigation": true,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      <DropdownItem
                        onClick={[Function]}
>
                        Edit
</DropdownItem>,
                      <DropdownItem
                        onClick={[Function]}
>
                        Delete
</DropdownItem>,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Edit",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Edit",
                      "type": [Function],
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "class",
                      "props": Object {
                        "children": "Delete",
                        "onClick": [Function],
                      },
                      "ref": null,
                      "rendered": "Delete",
                      "type": [Function],
                    },
                  ],
                  "type": [Function],
                },
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Dropdown
              additionLabel="Add "
              additionPosition="top"
              className="top right pointing dropdown"
              closeOnBlur={true}
              deburr={false}
              floated="right"
              icon="settings"
              minCharacters={1}
              noResultsMessage="No results found."
              openOnFocus={true}
              renderLabel={[Function]}
              searchInput="text"
              selectOnBlur={true}
              selectOnNavigation={true}
>
              <DropdownMenu>
                            <DropdownItem
                                          onClick={[Function]}
                            >
                                          Edit
                            </DropdownItem>
                            <DropdownItem
                                          onClick={[Function]}
                            >
                                          Delete
                            </DropdownItem>
              </DropdownMenu>
</Dropdown>,
            "minWidth": 768,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "additionLabel": "Add ",
              "additionPosition": "top",
              "children": <DropdownMenu>
                <DropdownItem
                                onClick={[Function]}
                >
                                Edit
                </DropdownItem>
                <DropdownItem
                                onClick={[Function]}
                >
                                Delete
                </DropdownItem>
</DropdownMenu>,
              "className": "top right pointing dropdown",
              "closeOnBlur": true,
              "deburr": false,
              "floated": "right",
              "icon": "settings",
              "minCharacters": 1,
              "noResultsMessage": "No results found.",
              "openOnFocus": true,
              "renderLabel": [Function],
              "searchInput": "text",
              "selectOnBlur": true,
              "selectOnNavigation": true,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <DropdownItem
                    onClick={[Function]}
>
                    Edit
</DropdownItem>,
                  <DropdownItem
                    onClick={[Function]}
>
                    Delete
</DropdownItem>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Edit",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Edit",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "children": "Delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Delete",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "cake": Object {
              "desc": "description of test cake",
              "image": "https://picsum.photos/400",
              "title": "Testy Cake",
            },
            "closeIcon": true,
            "closeOnDocumentClick": true,
            "onClose": [Function],
            "onSubmit": [Function],
            "open": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
